libs/platform/ebpf_platform.h

```diff
@@ -221,43 +219,43 @@ extern "C"
 
     /**
      * @brief Create an instance of a lock.
-     * @param[in] lock Pointer to memory location that will contain the lock.
+     * @param[out] lock Pointer to memory location that will contain the lock.
      */
     void
-    ebpf_lock_create(ebpf_lock_t* lock);
+    ebpf_lock_create(_Inout_ ebpf_lock_t* lock);
 
     /**
      * @brief Destroy an instance of a lock.
      * @param[in] lock Pointer to memory location that contains the lock.
      */
     void
-    ebpf_lock_destroy(ebpf_lock_t* lock);
+    ebpf_lock_destroy(_In_ ebpf_lock_t* lock);
 
     /**
      * @brief Acquire exclusive access to the lock.
      * @param[in] lock Pointer to memory location that contains the lock.
      * @param[out] state Pointer to memory location that contains state that
      *    needs to be passed to ebpf_lock_unlock.
      */
-    void
-    ebpf_lock_lock(ebpf_lock_t* lock, ebpf_lock_state_t* state);
+    _Requires_lock_not_held_(*lock) _Acquires_lock_(*lock) _IRQL_requires_max_(DISPATCH_LEVEL) _IRQL_saves_
+        _IRQL_raises_(DISPATCH_LEVEL) ebpf_lock_state_t ebpf_lock_lock(_In_ ebpf_lock_t* lock);
```

Comment:
since this API can be used from UM too, maybe these IRQL related SAL annotations should be moved to a separate header in kernel\ebpf_platform.h ?

Reply:
The IRQL related SAL gets stripped when compiled for user mode, so it has no effect.